util/store.go

```diff
@@ -0,0 +1,113 @@
+package util
+
+import (
+	"bytes"
+	"encoding/gob"
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/boltdb/bolt"
+)
+
+// Store is the parameter store database container.
+type Store struct {
+	name       string
+	bucketName []byte
+	db         *bolt.DB
+}
+
+// Initialize a new persistent key value store in os temp directory
+func NewStore(name string) (*Store, error) {
+	file := fmt.Sprintf("%s/%s.db", os.TempDir(), name)
```

Comment:
Weiß nicht ob TempDir ne gute Idee ist. Das OS behält sich idR vor hier auch bei Bedarf aufzuräumen.

In der Unix Welt würde ich diese Datei eher unter /var/evcc/[...].db oder so sehen.

Spannende Frage ist auch ob wir mehrere DBs machen oder ob nicht eine DB für alles ne gute Idee ist. Ich hatte das savings feature ja erstmal bewusst einfach gehalten (ein Zähler, keine Zeiträume, ...). Kann mir aber gut vorstellen, dass wir mal sowas wie ein Ladelog machen. Ich kenne aber auch die best practices bei dieser Bibliothek noch nicht.

Reply:
> Spannende Frage ist auch ob wir mehrere DBs machen oder ob nicht eine DB für alles ne gute Idee ist.

Du kannst innerhalb der DB mehrere "Buckets" definieren um Parameterkontexte zu unterscheiden. In der aktuellen Umsetzung habe ich DB = Bucket gesetzt. Ich kann es aber auch so erweitern, das man den Befehlen Put, Get und Delete einen Bucketnamen mitgeben kann.

Nur wenn dieser nicht gesetzt ist, kann ich db/file Namen = Bucketname setzen.